course-instructions.md

```diff
@@ -0,0 +1,760 @@
+
+tar_option_set(packages = c("dataRetrieval", "dplyr", "httr", "lubridate", "maps",
+                            "maptools", "RColorBrewer", "rgeos", "rgdal", "sp", "yaml"))
+
+# Load configuration files
+p0_targets_list <- list(
+  tar_target(map_config_yml, "configs/mapping.yml", format = "file"),
+  tar_target(map_config, yaml.load_file(map_config_yml)),
+  tar_target(wqp_config_yml, "configs/wqp_params.yml", format = "file")
+  tar_target(wqp_config, yaml.load_file(wqp_config_yml))
+)
+
+# Fetch data
+p1_targets_list <- list(
+  tar_target(huc_map, get_mutate_HUC8s(map_config)),
+  tar_target(phosphorus_lakes, get_wqp_data("phosphorus_lakes", wqp_config, map_config)),
+  tar_target(phosphorus_all, get_wqp_data("phosphorus_all", wqp_config, map_config)),
+  tar_target(nitrogen_lakes, get_wqp_data("nitrogen_lakes", wqp_config, map_config)),
+  tar_target(nitrogen_all, get_wqp_data("nitrogen_all", wqp_config, map_config)),
+  tar_target(arsenic_lakes, get_wqp_data("arsenic_lakes", wqp_config, map_config)),
+  tar_target(arsenic_all, get_wqp_data("arsenic_all", wqp_config, map_config)),
+  tar_target(temperature_lakes, get_wqp_data("temperature_lakes", wqp_config, map_config)),
+  tar_target(temperature_all, get_wqp_data("temperature_all", wqp_config, map_config)),
+  tar_target(secchi_lakes, get_wqp_data("secchi_lakes", wqp_config, map_config)),
+  tar_target(secchi_all, get_wqp_data("secchi_all", wqp_config, map_config)),
+)
+
+# Summarize the data in a plot
+p2_targets_list <- list(
+  tar_target(
+    multi_panel_constituents_png,
+    plot_huc_panel(
+      "figures/multi_panel_constituents.png", huc_map, map_config, 
+      arsenic_lakes, arsenic_all, nitrogen_lakes, nitrogen_all, 
+      phosphorus_lakes, phosphorus_all, secchi_lakes, secchi_all, 
+      temperature_lakes, temperature_all
+    ),
+    format = "file")
+)
+
+# Combine all targets into a single list
+c(p0_targets_list, p1_targets_list, p2_targets_list)
+```
+
+This makefile recipe generates a multipanel map, which colors [HUC8 watersheds](http://dep.wv.gov/WWE/getinvolved/sos/Documents/Basins/HUCprimer.pdf) according to how many sites within the watershed have data for various water quality constituents:
+![multi_panel_constituents](https://user-images.githubusercontent.com/2349007/82117369-18999280-9735-11ea-8365-e58742c5ff7e.png)
+
+---
+
+The `"figures/multi_panel_constituents.png"` figure takes a while to plot, so it is a somewhat "expensive" target to iterate on when it comes to style, size, colors, and layout (it takes 3 minutes to plot for me). But the plotting expense is dwarfed by the amount of time it takes to build each water quality data "object target", since `get_wqp_data` uses a web service that queries a large database and returns a result; the process of fetching the data can sometimes take over thirty minutes (`nitrogen_all` is a target that contains the locations of all of the sites that have nitrogen water quality data samples). 
```

Comment:
3 minutes doesn't seem too excessive to me for a multi-panel CONUS map.

Reply:
#29 I'll revisit this language - 3 min definitely isn't. However, it's a balancing act when training people!